app/controllers/components/course/controller_component_host.rb

```diff
@@ -66,6 +66,24 @@ def can_be_disabled?
   # Eager load all the components declared.
   eager_load_components(File.join(__dir__, '../'))
 
+  class << self
+    # Returns all components which can be disabled.
+    #
+    # @return [Array<Class>] array of disable-able components
+    def disableable_components
+      components.select(&:can_be_disabled?)
```

Comment:
Would we want to use a class variable here instead? I'm trying to read up to see if that is helpful, but I'm not sure if there are any gotchas. I think there aren't as long as the `can_be_disabled?` is hardcoded in our code, rather than dynamically set at the user level.